p2p/pex/pex_reactor.go

```diff
@@ -414,6 +421,8 @@ func (r *Reactor) SetEnsurePeersPeriod(d time.Duration) {
 
 // Ensures that sufficient peers are connected. (continuous).
 func (r *Reactor) ensurePeersRoutine() {
+	defer r.peersRoutineWg.Done()
+
 	var (
 		seed   = cmtrand.NewRand()
 		jitter = seed.Int63n(r.ensurePeersPeriod.Nanoseconds())
```

Comment:
At startup we sleep for this period. We should consider a select inside the if of line 434 so that we don't block `Stop()` while sleeping there.

Reply:
Why? It will make the code more complicated, but the chance of someone stopping the node right after start is super low, no?